packages/components/src/navigation/item/index.js

```diff
@@ -70,34 +64,23 @@ export default function NavigationItem( props ) {
 		onClick( event );
 	};
 	const icon = isRTL ? chevronLeft : chevronRight;
+	const baseProps = isText
+		? { as: 'div', ...restProps }
+		: { as: Button, href, onClick: onItemClick, ...restProps };
 
 	return (
-		<ItemUI className={ classes }>
+		<NavigationItemBase { ...props } className={ classes }>
 			{ children || (
-				<Button href={ href } onClick={ onItemClick } { ...restProps }>
-					{ title && (
-						<ItemTitleUI
-							className="components-navigation__item-title"
-							variant="body.small"
-							isRTL={ isRTL }
-							as="span"
-						>
-							{ title }
-						</ItemTitleUI>
-					) }
-
-					{ badge && (
-						<ItemBadgeUI
-							className="components-navigation__item-badge"
-							isRTL={ isRTL }
-						>
-							{ badge }
-						</ItemBadgeUI>
-					) }
+				<ItemBaseUI { ...baseProps }>
```

Comment:
I think we should update the styled components names to keep them consistent with where they are used.

For example, we use `ItemBaseUI` inside `NavigationItem` and `ItemUI` inside `NavigationItemBase`.
I understand the reasoning (and its history), but it's also counterintuitive.

Reply:
One other thing: what if we moved this `ItemBaseUI` into `NavigationItemBaseContent`?

Something like this:
```jsx
function NavigationItem( props ) {
  const baseContentProps = { title, badge, isRTL };

  const baseProps = isText
    ? { as: 'div', ...baseContentProps, ...restProps }
    : { as: Button, href, onClick: onItemClick, ...baseContentProps, ...restProps };

  return ( <NavigationItemBase>
    <NavigationItemBaseContent { ...baseProps } />
  </NavigationItemBase> );
}

function NavigationItemBaseContent( props ) {
  return ( <ItemBaseUI title={ title } badge={ badge } isRTL={ isRTL }>
    { title && <ItemTitleUI /> }
    { badge && <ItemBadgeUI /> }
  </ItemBaseUI> );
}
```